x-pack/plugins/observability/public/pages/alerts/components/alerts_search_bar.tsx

```diff
@@ -47,7 +48,7 @@ export function AlertsSearchBar({
 
   return (
     <SearchBar
-      indexPatterns={compatibleIndexPatterns}
+      indexPatterns={compatibleIndexPatterns as DataView[]}
```

Comment:
This is ok as far as the prop `indexPatterns`  exposed by `SearchBar` is the same type `DataView`
https://github.com/elastic/kibana/pull/129506/files#diff-c82e1b8d881f115132594175097c24185f00063c492eb44af979776de2218f53R42

Reply:
Let's merge this one as the `DataView` type is used in both places. Later on, If the type needs to be changed, it will be handled in a different PR.